hack/generate-manifest.sh

```diff
@@ -104,46 +110,86 @@ elif ! $KUSTOMIZE version > /dev/null 2>&1; then
     exit 1
 fi
 
-KUSTOMIZATION_DIR=$THIS_DIR/../build/yamls/overlays
+KUSTOMIZATION_DIR=$THIS_DIR/../build/yamls
 
-TMP_DIR=$(mktemp -d $KUSTOMIZATION_DIR/$MODE.XXXXXXXX)
+TMP_DIR=$(mktemp -d $KUSTOMIZATION_DIR/overlays.XXXXXXXX)
 
 pushd $TMP_DIR > /dev/null
 
-touch kustomization.yml
-$KUSTOMIZE edit add base ../$MODE
+BASE=../../base
 
-if [ "$MODE" == "dev" ]; then
-    # nothing to do for now, everything is taken care of in the overlay kustomization.yml.
-    :
+OVS_DATAPATH_TYPE=system
+if $KIND; then
+    OVS_DATAPATH_TYPE=netdev
 fi
-
-if [ "$MODE" == "release" ]; then
-    $KUSTOMIZE edit set image antrea=$IMG_NAME:$IMG_TAG
+# do all ConfigMap edits
+mkdir configMap && cd configMap
```

Comment:
I would do this only for KIND, but keep the original default value in antrea-agent.tpl.conf for other cases.

Reply:
I considered both. I went with this option because I thought it was more simple / robust to have `ovsDatapathType: <OVS_DATAPATH_TYPE>` in the template file and do a simple `sed` for the `<OVS_DATAPATH_TYPE>` token unconditionally, rather than leave `#ovsDatapathType: system` in the template and use a more "convoluted" regex (for the transformation `#ovsDatapathType: system` -> `ovsDatapathType: netdev`) when Kind is enabled (the same reasoning applies to ipsec).

That being said, I don't have too strong of an opinion so I'm open to making that change if you prefer using an "implicit" default for the regular case.

Reply:
I was thinking to change #ovsDatapathType: system" to "ovsDatapathType: netdev" with sed. Prefer to keep the template with all default options/values. Also we will have other cases to change an option (e.g. IPSec, encap, etc.), I would not maintain all such options differently in template.

Reply:
Actually 1) do we expect users / developers to be able to modify options - such as MTU - in agent.tpl.conf before generating the YAML or 2) do we expect them to make changes in the generated YAML only.

If 1), I think the current solution is better. If 2), then we don't need the agent.tpl.conf template file at all: I can just keep agent.conf as it is and use `sed` rules such as `#ovsDatapathType: system` -> `ovsDatapathType: netdev`.

Reply:
I assume users will not modify the agent.tpl.conf or agent.conf, and developers modify it only when changing the template (that should be merged to git) - do not feel it is simpler than editting the generated yaml to make any change for deployment.

I am fine with either agent.conf, or agent.conf.tpl or agent.conf.in - the latter to indicate it might be changed during building.

Reply:
I made the change. PTAL.